src/include/duckdb/main/query_profiler.hpp

```diff
@@ -34,14 +34,19 @@ class PhysicalOperator;
 class SQLStatement;
 
 struct OperatorInformation {
-	explicit OperatorInformation(double time_p = 0, idx_t elements_returned_p = 0, idx_t elements_scanned_p = 0,
-	                             idx_t result_set_size_p = 0)
-	    : time(time_p), elements_returned(elements_returned_p), result_set_size(result_set_size_p) {
+	explicit OperatorInformation(double time_p = 0, idx_t elements_returned_p = 0, idx_t result_set_size_p = 0,
+	                             idx_t system_peak_buffer_manager_memory_p = 0,
+	                             idx_t system_peak_temp_directory_size_p = 0)
+	    : time(time_p), elements_returned(elements_returned_p), result_set_size(result_set_size_p),
+	      system_peak_buffer_manager_memory(system_peak_buffer_manager_memory_p),
+	      system_peak_temp_directory_size(system_peak_temp_directory_size_p) {
 	}
 
 	double time;
 	idx_t elements_returned;
 	idx_t result_set_size;
+	idx_t system_peak_buffer_manager_memory;
+	idx_t system_peak_temp_directory_size;
```

Comment:
Given that we keep likely keep expanding these, and since this PR also adds two additional parameters to the constructor: should we maybe move these into a `struct`? Something like `OperatorInfoMetrics`. 🤔

Reply:
It's already `struct OperatorInformation`, but I agree that this constructor is getting out of hand. I checked, and nowhere in the code base are we using the constructor; we always use the defaults. So instead, I've removed the arguments from the constructor, and made it so the values are initialized within the structure, which I think is a lot cleaner.

Reply:
Yes, great! 👍